### PR TITLE
KEP-3926: refactor: extend newTransformTest to enable RBAC

### PR DESCRIPTION
**Pull request by tkashem**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

https://github.com/kubernetes/kubernetes/pull/127513 enables RBAC for its tests

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

KEP: https://github.com/kubernetes/enhancements/pull/3927

```docs

```

### CODE REVIEW

File: test/integration/controlplane/transformation/transformation_test.go [hunk 1]
```diff
@@ -286,14 +288,18 @@ func (e *transformTest) getRawSecretFromETCD() ([]byte, error) {
...
 	if e.transformerConfig != "" {
-		return []string{
+		options = append(options, []string{
 			"--encryption-provider-config", filepath.Join(e.configDir, encryptionConfigFileName),
 			fmt.Sprintf("--encryption-provider-config-automatic-reload=%v", reload),
-			"--disable-admission-plugins", "ServiceAccount"}
+			"--disable-admission-plugins", "ServiceAccount"}...)
+	}
+	if e.enableRBAC {
+		options = append(options, "--authorization-mode=RBAC")
```
Comment (tkashem) [inline test/integration/controlplane/transformation/transformation_test.go:299]: let me know RBAC to be enabled for all existing tests by default

Comment (enj) [reply test/integration/controlplane/transformation/transformation_test.go:299]: @tkashem yeah, just enable it always.  The diff should be small enough to simply include in #127513

**Comment by tkashem** [comment]
/test pull-kubernetes-e2e-kind-ipv6

```
Kubernetes e2e suite: [It] [sig-network] Services should be able to connect to terminating and unready endpoints if PublishNotReadyAddresses is true 
```

File: test/integration/controlplane/transformation/transformation_test.go (hunk 1 above)
Comment (tkashem) [reply test/integration/controlplane/transformation/transformation_test.go]: yes, enabled, and the tests are green so far.

**Comment by tkashem** [comment]
https://github.com/kubernetes/kubernetes/pull/127513 merged with this commit

/close

**Pull request closed without merging**